### PR TITLE
Sticky footer and with editing hoc uts

### PR DESCRIPTION
**Pull request by MichelML**

### CODE REVIEW

File: src/hoc/withEditing/withEditing.tsx [hunk 1]
```diff
@@ -21,7 +23,9 @@ export interface IWithEditDispatchProps {
     toggleDirtyComponent: (isDirty: boolean) => void;
 }
 
-export const withEditing = (config: IWithEditing) => (Component: React.ComponentClass<any, any>): React.ComponentClass<any, any> => {
+export interface IWithEditingProps extends IWithEditStateProps, IWithEditDispatchProps {}
+
+export const withEditing = (config: IWithEditing) => (Component: React.ComponentClass): React.ComponentClass => {
     const mapStateToProps = (state: IReactVaporState): IWithEditStateProps => ({
         isDirty: !!_.contains(state.dirtyComponents, config.id),
```
Comment (pochretien) [inline src/hoc/withEditing/withEditing.tsx]: add a selector

File: src/hoc/withEditing/withEditing.tsx [hunk 2]
```diff
@@ -31,19 +35,23 @@ export const withEditing = (config: IWithEditing) => (Component: React.Component
     });
 
     @ReduxConnect(mapStateToProps, mapDispatchToProps)
-    class ComponentWithEditing extends React.Component<any, any> {
+    class ComponentWithEditing extends React.Component<IWithEditingProps> {
+        componentDidMount() {
+            callIfDefined(this.props.toggleDirtyComponent, config.isDirty);
```
Comment (pochretien) [inline src/hoc/withEditing/withEditing.tsx]: toggleDirtyComponent cant be undefined. you can remove callIfDefined

File: src/hoc/withEditing/withEditing.tsx [hunk 3]
```diff
@@ -31,19 +35,23 @@ export const withEditing = (config: IWithEditing) => (Component: React.Component
...
+
         componentWillUnmount() {
             callIfDefined(this.props.toggleDirtyComponent, false);
         }
 
         render() {
             return (
-                <div className={styles.stickyFooterPadding}>
-                    <Component {..._.omit(this.props, 'isDirty', 'footerChildren')}>
+                <div>
```
Comment (pochretien) [inline src/hoc/withEditing/withEditing.tsx]: we can use fragment ? I dont know if react-vapor support this new feature

File: src/components/stickyFooter/StickyFooter.scss [hunk 4]
```diff
@@ -16,30 +11,20 @@ $stickyFooterHeight: 58px;
...
     border-radius: 0 0 $border-radius $border-radius;
     padding: $modal-footer-small-padding;
 
-
-    margin-bottom: -$navigation-width;
-
-    transform: translate3d(0, 0, 0);
-
-    transition: margin-bottom $navigation-toggle-duration ease-in-out, width $navigation-toggle-duration ease-in-out;
+    transform: translate3d(0, 100px, 0);
```
Comment (GermainBergeron) [inline src/components/stickyFooter/StickyFooter.scss]: Can you use 100% instead of 100px?

File: src/hoc/withEditing/withEditing.tsx (hunk 3 above)
Comment (gdostie) [reply src/hoc/withEditing/withEditing.tsx]: React-Vapor supports it, but the admin doesn't :crying_cat_face:

File: src/components/stickyFooter/StickyFooter.scss (hunk 4 above)
Comment (MichelML) [reply src/components/stickyFooter/StickyFooter.scss]: on it

File: src/hoc/withEditing/withEditing.tsx (hunk 2 above)
Comment (MichelML) [reply src/hoc/withEditing/withEditing.tsx]: removed

**Pull request merged**